src/distribute_kv_cache/rpc/server.rs

```diff
@@ -0,0 +1,563 @@
+                return Err(err);
+            }
+        }
+
+        let buffer: &mut BytesMut = unsafe { &mut *self.req_buf.get() };
+        let req_header = ReqHeader::decode(buffer)?;
+        debug!("Received request header: {:?}", req_header);
+
+        Ok(req_header)
+    }
+
+    /// Recv request body from the stream
+    pub async fn recv_len(&self, len: u64) -> Result<(), RpcError> {
+        let start = tokio::time::Instant::now();
+        let mut req_buffer: &mut BytesMut = unsafe { &mut *self.req_buf.get() };
+        utils::ensure_buffer_len(req_buffer, u64_to_usize(len));
+        let start_1 = start.elapsed();
+        debug!("Server resize buffer to size {:?} cost: {:?}", len, start_1);
+
+        let reader = self.get_stream_mut();
+        match read_exact_timeout!(reader, &mut req_buffer, self.timeout_options.read_timeout).await
+        {
+            Ok(size) => {
+                debug!("Received request body: {:?}", size);
+                Ok(())
+            }
+            Err(err) => {
+                debug!("Failed to receive request: {:?}", err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Recv huge request body from the stream
+    pub async fn recv_len_pass_in_len(
+        &self,
+        len: u64,
+        req_buffer: &mut BytesMut,
+    ) -> Result<(), RpcError> {
+        let start = tokio::time::Instant::now();
+        utils::ensure_buffer_len(req_buffer, u64_to_usize(len));
+        let start_1 = start.elapsed();
+        debug!("Server resize buffer to size {:?} cost: {:?}", len, start_1);
+
+        let reader = self.get_stream_mut();
+        match read_exact_timeout!(reader, req_buffer, self.timeout_options.read_timeout).await {
+            Ok(size) => {
+                debug!("Received request body: {:?}", size);
+                Ok(())
+            }
+            Err(err) => {
+                debug!("Failed to receive request: {:?}", err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Send response to the stream
+    /// The response is a byte array, contains the response header and body.
+    pub async fn send_response(&self, resp: &[u8]) -> Result<(), RpcError> {
+        let writer = self.get_stream_mut();
+        match write_all_timeout!(writer, resp, self.timeout_options.write_timeout).await {
+            Ok(()) => {
+                debug!("Sent response successfully: {:?}", resp.len());
+                Ok(())
+            }
+            Err(err) => {
+                debug!("Failed to send response: {:?}", err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Send vectored response to the stream
+    /// The response is a byte array, contains the response header and body.
+    pub async fn send_vectored_response(&self, resp: &[IoSlice<'_>]) -> Result<usize, RpcError> {
```

Comment:
there's no usage in this PR. Please make sure the caller check the written size in the future PRs